adguard_cb/src/main/java/com/adguard/android/service/FilterServiceImpl.java

```diff
@@ -259,7 +265,19 @@ public void applyNewSettings() {
         List<String> rules = getAllEnabledRules(true);
         Set<String> userRules = getUserRules();
         if (!userRules.isEmpty()) {
-            rules.addAll(userRules);
+            for (String userRule : userRules) {
+                userRule = StringUtils.trim(userRule);
+                
+                if (StringUtils.isNotBlank(userRule) &&
```

Comment:
Wouldn't it be better to have a "validateRuleText" method instead.

Also, it'd be better to validate main filter rules as well.